### DISCUSSION TITLE
Testing links

### DISCUSSION THREAD

**Discussion by PurpleBooth** [vercel/next.js#39958] [Help] [answered]
When using other frameworks such as CRA, testing links is trivial, but I am struggling to work out how to do it without spinning up a whole browser with next, I assume it's equaly as easy with Next.js and I am just missing something.

Does anyone have any links or advice? (I am using Jest)

**Comment by icyJoseph** [comment, marked as answer]
You can use https://www.npmjs.com/package/next-router-mock, which makes it work pretty well.

That being said, some tests that involve navigation and such, probably should be run in a headless browser, so called E2E.

**Reply by PurpleBooth** [reply]
Nice thanks! I will give that a go. I have Playwright tests too, but I would rather test lower down if it's possible.
